twoliter/embedded/imghelper

```diff
@@ -0,0 +1,398 @@
+VERITY_HASH_ALGORITHM=sha256
+VERITY_DATA_BLOCK_SIZE=4096
+VERITY_HASH_BLOCK_SIZE=4096
+
+install_ca_certs() {
+  local root_mount
+  root_mount="${1:?}"
+  install -d "${root_mount}${CA_CERTS_INSTALLDIR}"
+  if [[ -s "${CA_CERTS_BUNDLE}" ]]; then
+    install -p -m 0644 \
+      "${CA_CERTS_BUNDLE}" \
+      "${root_mount}${CA_CERTS_INSTALLDIR}"
+  elif [[ -s "${CA_CERTS_DEFAULT_BUNDLE}" ]]; then
+    install -p -m 0644 \
+      "${CA_CERTS_DEFAULT_BUNDLE}" \
+      "${root_mount}${CA_CERTS_INSTALLDIR}"
+    install -p -m 0644 \
+      -D "${CA_CERTS_DEFAULT_LICENSES}"/* \
+      -t "${root_mount}/usr/share/licenses/ca-certificates"
+  else
+    echo "no CA certificates found" >&2
+    exit 1
+  fi
+  mkdir -p "${root_mount}${CROSS_TMPFILESDIR}"
+  cat <<EOF >>"${root_mount}${CROSS_TMPFILESDIR}/ca-certificates.conf"
+C /etc/pki/tls/certs/ca-bundle.crt - - - -
+L /etc/ssl/certs - - - - ../pki/tls/certs
+EOF
+}
+
+# shim expects the following data structure in `.vendor_cert`:
+#
+# struct {
+#   uint32_t vendor_authorized_size;
+#   uint32_t vendor_deauthorized_size;
+#   uint32_t vendor_authorized_offset;
+#   uint32_t vendor_deauthorized_offset;
+# } cert_table;
+#
+cert_table() {
+  local input output size offset uint32_t
+  input="${1:?}"
+  output="${2:?}"
+  size="$(stat -c %s "${input}")"
+  rm -f "${output}"
+  # The cert payload is offset by four 4-byte uint32_t values in the header.
+  offset="$((4 * 4))"
+  for n in "${size}" 0 "${offset}" "$((size + offset))"; do
+    printf -v uint32_t '\\x%02x\\x%02x\\x%02x\\x%02x' \
+      $((n & 255)) $((n >> 8 & 255)) $((n >> 16 & 255)) $((n >> 24 & 255))
+    # shellcheck disable=SC2059  # Variable contains a pattern
+    printf "${uint32_t}" >>"${output}"
+  done
+  cat "${input}" >>"${output}"
+  # Zero-pad the output to the expected section size. Otherwise a subsequent
+  # `objcopy` operation on the same section might fail to replace it, if the
+  # new vendor certificate is larger than this one.
+  truncate -s 4096 "${output}"
+}
+
+# Helper function to log the object layout before and after changes.
+objdumpcopy() {
+  local obj objdump objcopy
+  obj="${1:?}"
+  shift
+  objdump="${ARCH}-bottlerocket-linux-gnu-objdump"
+  objcopy="${ARCH}-bottlerocket-linux-gnu-objcopy"
+  "${objdump}" -h "${obj}"
+  "${objcopy}" "${@}" "${obj}"
+  "${objdump}" -h "${obj}"
+}
+
+sbsetup() {
+  local pedb sb_key_source
+  local -n shim_sign_key code_sign_key
+  shim_sign_key="${1:?}"
+  code_sign_key="${2:?}"
+
+  # For an AWS profile, we expect a config file for the PKCS11
+  # helper. Otherwise, there should be a local key and cert.
+  if [[ -s "${HOME}/.config/aws-kms-pkcs11/config.json" ]]; then
+    # Set AWS environment variables from build secrets, if present.
+    for var in AWS_ACCESS_KEY_ID AWS_SECRET_ACCESS_KEY AWS_SESSION_TOKEN; do
+      val="${var,,}"
+      val="${HOME}/.aws/${val//_/-}.env"
+      [[ -s "${val}" ]] || continue
+      declare -x "${var}=$(cat "${val}")"
+    done
```

Comment:
Let's split this into two helper functions - `sbsetup_aws_profile` and `sbsetup_local_profile` - that we call depending on which branch we take. Effectively they both have the same interface - populate the `shim_sign_key` and `code_sign_key` args - so they can just take those by reference.